services/swagger.js

```diff
@@ -21,10 +21,10 @@ var httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
 
 var responseModelValidationLevel;
 var polymorphicValidation;
+var refCompilation;
 
-exports.init = function (logger, config, callback) {
+exports.init = function (logger, config, refCompiler, callback) {
```

Comment:
What is the benefit of having the ref compiler as its own service?
How could it be used outside of server startup?

I think it would be better to have it simply required as a regular node module from the `swagger` service and called as part of `init` (just as it is now).